tests/realtime/redisPublisher.js

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) 2017, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * tests/realtime/setupSocketIO.js
+ */
+'use strict';
+
+const expect = require('chai').expect;
+const tu = require('../testUtils');
+const u = require('./utils');
+const Subject = tu.db.Subject;
+const Aspect = tu.db.Aspect;
+const Sample = tu.db.Sample;
+const publisher = require('../../realtime/redisPublisher');
+const sampleEvent = require('../../realtime/constants').events.sample;
+
+describe('redis Publisher', () => {
+  const subjectNA = { name: `${tu.namePrefix}NorthAmerica`, isPublished: true };
+  const subjectSA = { name: `${tu.namePrefix}SouthAmerica`, isPublished: true };
+
+  const samp = { value: 10 };
+  let sampId;
+  let ipar;
+  const humidity = {
+    name: `${tu.namePrefix}humidity`,
+    timeout: '60s',
+    isPublished: true,
+  };
+  before((done) => {
+    Subject.create(subjectNA)
+    .then((subj) => {
+      ipar = subj.id;
+      return Aspect.create(humidity);
+    })
+    .then((asp) => {
+      samp.subjectId = ipar;
+      samp.aspectId = asp.id;
+      return Sample.create(samp);
+    })
+    .then((s) => {
+      sampId = s.id;
+      return Subject.create(subjectSA);
+    })
+    .then(() => done())
+    .catch(done);
+  });
+  after(u.forceDelete);
+  describe('publishSample function tests: ', () => {
+    it('with EventType argument: sample should be published with subject  ' +
```

Comment:
It is the type of the event that is published. There is more information about it in publisher.js